src/custom/hooks/web3.ts

```diff
@@ -130,10 +130,29 @@ export function useEagerConnect() {
   }, [active])
 
   useEffect(() => {
-    // add beforeunload event listener on initial component mount
+    // Check if current connector is of type WalletConnect
+    // Fix for this https://github.com/gnosis/cowswap/issues/1923
+    if (connector instanceof WalletConnectConnector) {
+      const walletConnect = connector.walletConnectProvider.signer.connection.wc
+
+      // Listen on disconnect events directly on WalletConnect client and close the connection
+      // Important in case the connection is closed from the wallet side after the page is refreshed
+      walletConnect.on('disconnect', (error: Error) => {
+        if (error) {
+          console.error('[WalletConnectConnector] Error during disconnect:', error)
+        } else {
+          connector.close()
+          localStorage.removeItem(STORAGE_KEY_LAST_PROVIDER)
+        }
+      })
+    }
+  }, [connector])
+
+  useEffect(() => {
+    // Add beforeunload event listener on initial component mount
     window.addEventListener('beforeunload', handleBeforeUnload)
 
-    // remove beforeunload event listener on component unmount
+    // Remove beforeunload event listener on component unmount
```

Comment:
is this some new grammarnazi lint rule or is this custom? seems very unnecessary